### PR TITLE
[21334] Handle nullptr in parseXMLStructDynamicType

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This PR handles a nullptr in `parseXMLStructDynamicType`

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- N/A Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- __NO__ Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- __NO__ Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- **N/A** If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/xmlparser/XMLDynamicParser.cpp [hunk 1]
```diff
@@ -982,9 +982,17 @@ XMLP_ret XMLParser::parseXMLStructDynamicType(
     const char* baseType = p_root->Attribute(BASE_TYPE);
     if (baseType != nullptr)
     {
-        DynamicTypeBuilder::_ref_type parent_type_builder;
+        DynamicTypeBuilder::_ref_type parent_type_builder = nullptr;
+
```
Comment (JesusPoderoso) [inline src/cpp/xmlparser/XMLDynamicParser.cpp:986]: Uncrustify issue
```suggestion
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
NIT

File: src/cpp/xmlparser/XMLDynamicParser.cpp (hunk 1 above)
Comment (elianalf) [reply src/cpp/xmlparser/XMLDynamicParser.cpp:986]: don't you think it's clearer this way?

Comment (JesusPoderoso) [reply src/cpp/xmlparser/XMLDynamicParser.cpp:986]: I think so but uncrustify does not 😕

**Comment by elianalf** [comment]
CI passed here [d2eecdb](https://github.com/eProsima/Fast-DDS/pull/5080/commits/d2eecdb9d42115a4ac741d7844bb6d6ba1c3b458)

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**